app/src/main/java/org/liberty/android/fantastischmemo/dao/HistoryDaoImpl.java

```diff
@@ -0,0 +1,89 @@
+package org.liberty.android.fantastischmemo.dao;
+
+
+import com.j256.ormlite.support.ConnectionSource;
+import com.j256.ormlite.table.DatabaseTableConfig;
+
+import org.liberty.android.fantastischmemo.common.AnyMemoDBOpenHelper;
+import org.liberty.android.fantastischmemo.entity.History;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Created by SnegaT on 2018-04-05.
+ */
+
+public class HistoryDaoImpl extends AbstractHelperDaoImpl<History, Integer> implements HistoryDao {
+
+    private AnyMemoDBOpenHelper helper = null;
+
+    protected HistoryDaoImpl(ConnectionSource connectionSource, DatabaseTableConfig<History> config) throws SQLException {
+        super(connectionSource, config);
+    }
+
+    public HistoryDaoImpl(ConnectionSource connectionSource, Class<History> clazz) throws SQLException {
+        super(connectionSource, clazz);
+    }
+
+    @Override
+    public AnyMemoDBOpenHelper getHelper() {
+        if (helper != null) {
+            return helper;
+        } else {
+            throw new IllegalStateException("Must set the helper in order to use.");
+        }
+    }
+
+    @Override
+    public void setHelper(AnyMemoDBOpenHelper helper) {
+        this.helper = helper;
+    }
+
+
+    public int create(History history) {
+        int res = super.create(history);
+        return res;
+    }
+
+    public int delete(History history) {
+        if (history != null) {
+            int res = super.delete(history);
+        }
+        return 0;
+    }
+
+    @Override
+    public List<History> getHistoryForDB(String dbPath) {
+        AnyMemoDBOpenHelper dbHelper = getHelper();
+        return dbHelper.getHistoryForDB(dbPath);
+    }
+
+    @Override
+    public History insertHistory(History hist) {
```

Comment:
Should this method be void?

Reply:
I thought it would be meaningful to have it return the history object we are inserting. If you think it should be null, can be changed.

Reply:
Nah its fine.